### DISCUSSION TITLE
Meta Char * does not iterate over folder images in this command

### DISCUSSION THREAD

**Discussion by spicyBeefu** [ImageMagick/ImageMagick#3621] [Help] [answered]
Forgive me being a beginner with this but I can't quite grasp why a single specified image will work but the * meta character cannot in the following command (For Windows7, IM 7.0.10-0). My intention is to repeatedly bleed color into alpha for every image in the folder and doing so in one command line is preferable. Any ideas?

works:
`convert ( 0001.png ( +clone -channel RGBA -blur 0x5 ) -compose DstOver -composite ) 0001.png -compose CopyOpacity -composite out.png`

returns single image without alpha:
`convert ( *.png ( +clone -channel RGBA -blur 0x5 ) -compose DstOver -composite ) *.png -compose CopyOpacity -composite out.png`

**Comment by snibgo** [comment, marked as answer]
"*.png" expands to all the PNG files in your current directory. So your command reads all the PNG files, then clones just the last one, and blurs that, and does a composite, then re-reads all the PNG files. That isn't what you want.

I would use a "for" loop. Type "help for" at the Windows command line. Like this:
```
for %X in (*.png) do convert ^( %X ^( +clone -channel RGBA -blur 0x5 ^) -compose DstOver -composite ^) %X -compose CopyOpacity -composite out.png
```
This runs "convert" once for each PNG file.

**Reply by spicyBeefu** [reply]
Thank you for the light speed reply (and great explanation), worked perfectly!
